crates/aptos/src/move_tool/template.rs

```diff
@@ -0,0 +1,280 @@
+                    std::env::temp_dir().join(format!("{}_{}", directory_name, time_millis));
+
+                git_download_custom_template(&custom_template_dir, git_url)?;
+
+                self.render_template_dir(custom_template_dir)?;
+            }
+        }
+        Ok(())
+    }
+}
+
+impl NewPackage {
+    fn package_name(&self) -> anyhow::Result<String> {
+        let package_name = match &self.name {
+            Some(name) => name.clone(),
+            None => self.package_dir.to_package_name(),
+        };
+        Ok(package_name)
+    }
+
+    fn render_empty_template(&self) -> anyhow::Result<()> {
+        let package_dir = self.package_dir.as_ref();
+        let package_name = self.package_name()?;
+
+        self.framework_package_args.init_move_dir(
+            package_dir,
+            &package_name,
+            BTreeMap::default(),
+            PromptOptions::default(),
+        )?;
+        fs::create_dir(package_dir.join("tests"))?;
+        Ok(())
+    }
+
+    fn render_template_dir(&self, template_dir_path: PathBuf) -> anyhow::Result<()> {
+        let package_dir = self.package_dir.as_ref();
+        let package_name = self.package_name()?;
+
+        let dot_git_path = template_dir_path.join(".git");
+        let template_fs_items = WalkDir::new(&template_dir_path)
+            .into_iter()
+            .filter_entry(move |entry| entry.path() != dot_git_path)
+            .filter_map(|entry| entry.ok())
+            .map(|entry| entry.into_path());
+
+        let renderer = Handlebars::new();
+        let template_context = json!({ "package_name": package_name });
+
+        let start_dir_pat = template_dir_path.to_str().unwrap();
+        for source_path in template_fs_items {
+            let subpath = source_path
+                .to_string_lossy()
+                .to_string()
+                .trim_start_matches(&start_dir_pat)
+                .trim_matches('/')
+                .to_string();
+
+            let dest_path = package_dir.join(subpath);
+            if dest_path.exists() {
+                continue;
+            }
+
+            if source_path.is_dir() {
+                fs::create_dir(dest_path)
+                    .map_err(|err| anyhow!("Create dir: {err}. {source_path:?}"))?;
+                continue;
+            }
+
+            if let Some(ext) = source_path.extension() {
+                if ext == "move" || ext == "toml" {
+                    // interpolate
+                    let contents = fs::read_to_string(source_path)?;
+                    let rendered_contents =
+                        renderer.render_template(&contents, &template_context)?;
+                    fs::write(dest_path, rendered_contents)?;
+                    continue;
+                }
+            }
+            // copy as-is
+            fs::copy(source_path, dest_path)?;
+        }
+        Ok(())
+    }
+}
+
+const GIT_APTOS_TEMPLATES_URL: &str = "https://github.com/mkurnikov/aptos-templates.git";
```

Comment:
this should be in aptos and I think it should be ideally be by branch and maybe commit

Reply:
It's already by commit, it works similarly to how dependencies work now.